crates/matrix-sdk/tests/integration/client.rs

```diff
@@ -583,3 +583,142 @@ async fn cross_signing_status() {
 
     server.verify().await;
 }
+
+#[cfg(feature = "e2e-encryption")]
+#[async_test]
+async fn encrypt_room_event() {
+    use std::sync::Arc;
+
+    use ruma::events::room::encrypted::RoomEncryptedEventContent;
+
+    let (client, server) = logged_in_client().await;
+    let user_id = client.user_id().unwrap();
+
+    Mock::given(method("POST"))
+        .and(path("/_matrix/client/r0/keys/query"))
+        .respond_with(ResponseTemplate::new(200).set_body_json(json!({
+            "device_keys": {
+                user_id: {}
+            }
+        })))
+        .mount(&server)
+        .await;
+
+    mock_sync(&server, &*test_json::SYNC, None).await;
+    client
+        .sync_once(SyncSettings::default())
+        .await
+        .expect("We should be able to performa an initial sync");
+
+    let room =
+        client.get_room(&DEFAULT_TEST_ROOM_ID).expect("We should know about our default room");
+
+    Mock::given(method("GET"))
+        .and(path_regex(r"^/_matrix/client/r0/rooms/.*/state/m.room.encryption/"))
+        .and(header("authorization", "Bearer 1234"))
+        .respond_with(ResponseTemplate::new(200).set_body_json(json!({
+            "algorithm": "m.megolm.v1.aes-sha2",
+            "rotation_period_ms": 604800000,
+            "rotation_period_msgs": 100
+        })))
+        .mount(&server)
+        .await;
+
+    assert!(
+        room.is_encrypted().await.expect("We should be able to check if the room is encrypted"),
+        "The room should be encrypted"
+    );
+
+    Mock::given(method("GET"))
+        .and(path_regex("/_matrix/client/r0/rooms/!SVkFJHzfwvuaIEawgC:localhost/members"))
+        .and(header("authorization", "Bearer 1234"))
+        .respond_with(ResponseTemplate::new(200).set_body_json(json!({ "chunk": []})))
+        .mount(&server)
+        .await;
+
+    let event_content = Arc::new(std::sync::Mutex::new(None));
+
+    let event_content_matcher = {
+        let event_content = event_content.to_owned();
+        move |request: &wiremock::Request| {
+            let path_segments =
+                request.url.path_segments().expect("The URL should be able to be a base");
+
+            let event_type = path_segments
+                .last()
+                .expect("The path should have a event type as the last segment")
```

Comment:
the path ends with the event id, so it's not quite the event type (but it would be, if the event id is popped)

Reply:
Right sorry